core/storage/trie/polkadot_trie_db/polkadot_codec.cpp

```diff
@@ -43,78 +43,64 @@ namespace kagome::storage::trie {
 
   inline common::Buffer ushortToBytes(uint16_t b) {
     common::Buffer out(2, 0);
-    out[0] = (b >> 8u) & 0xffu;
-    out[1] = b & 0xffu;
+    out[1] = (b >> 8u) & 0xffu;
+    out[0] = b & 0xffu;
     return out;
   }
 
   common::Buffer PolkadotCodec::nibblesToKey(const common::Buffer &nibbles) {
-    const auto nibbles_size = nibbles.size();
-    if (nibbles_size == 0) {
-      return {};
-    }
-
-    if (nibbles_size == 1 && nibbles[0] == 0) {
-      return {0};
-    }
-
-    // if nibbles_size is odd, then allocate one more item
-    const size_t size = (nibbles_size + 1) / 2;
-    common::Buffer out(size, 0);
-
-    size_t iterations = size;
-
-    // if number of nibbles is odd, then iterate even number of times
-    bool nimbles_size_odd = nibbles_size % 2 != 0;
-    if (nimbles_size_odd) {
-      --iterations;
-    }
-
-    for (size_t i = 0; i < iterations; ++i) {
-      out[i] = collectByte(nibbles[2 * i], nibbles[2 * i + 1]);
-    }
-
-    // if number of nibbles is odd, then implicitly add 0 as very last nibble
-    if (nimbles_size_odd) {
-      out[iterations] = collectByte(nibbles[2 * iterations], 0);
+    Buffer res;
+    if (nibbles.size() % 2 == 0) {
+      res = Buffer(nibbles.size() / 2, 0);
+      for (size_t i = 0; i < nibbles.size(); i += 2) {
+        res[i / 2] = (nibbles[i] << 4 & 0xf0) | (nibbles[i + 1] & 0xf);
+      }
+    } else {
+      res = Buffer(nibbles.size() / 2 + 1, 0);
+      res[0] = nibbles[0];
+      for (size_t i = 2; i < nibbles.size(); i += 2) {
+        res[i / 2] = (nibbles[i - 1] << 4 & 0xf0) | (nibbles[i] & 0xf);
+      }
     }
-
-    return out;
+    return res;
   }
 
   common::Buffer PolkadotCodec::keyToNibbles(const common::Buffer &key) {
-    auto nibbles = key.size() * 2;
-
-    // if last nibble in `key` is 0
-    bool last_nibble_0 = !key.empty() && (highNibble(key[key.size() - 1]) == 0);
-    if (last_nibble_0) {
-      --nibbles;
+    if (key.empty()) {
+      return {};
     }
-
-    common::Buffer out(nibbles, 0);
-    for (size_t i = 0u, size = nibbles / 2; i < size; i++) {
-      out[2 * i] = lowNibble(key[i]);
-      out[2 * i + 1] = highNibble(key[i]);
+    if (key.size() == 1 && key[0] == 0) {
+      return {0, 0};
     }
 
-    if (last_nibble_0) {
-      out[nibbles - 1] = lowNibble(key[key.size() - 1]);
+    auto l = key.size() * 2;
+    Buffer res(l, 0);
+    for (size_t i = 0; i < key.size(); i++) {
+      res[2 * i] = key[i] / 16;
+      res[2 * i + 1] = key[i] % 16;
```

Comment:
```suggestion
      res[2 * i + 1] = (key[i] >> 8) & 0xff;
```

Reply:
https://godbolt.org/z/m8qgFm
a=b, c=d, but b and d will be much faster (no / or %)

Reply:
Again, I don't know what is supposed to do here, but x%16 is the value represented by last 4 bits, so x%16 === x & 0b1111 === x & 0xf

Reply:
probably (x&0xFu) looks better